libraries/botbuilder-ai/src/luisRecognizerOptionsV3.ts

```diff
@@ -158,7 +176,9 @@ export class LuisRecognizerV3 extends LuisRecognizerInternal {
             result.entities[MetadataKey] = result.entities[MetadataKey] ? result.entities[MetadataKey] : {};
         }
 
-        this.emitTraceInfo(context, response.prediction, result, options);
+        if (context !== null) {
```

Comment:
Generally using `!= null` is better as both `null == null` and `undefined == null` are `true`. If you really just want to filter `null` and not `undefined` you would use `!==`.

Reply:
This was intentional since the only time it _should_ be null is from [this call](https://github.com/microsoft/botbuilder-js/pull/3690/files#diff-c3651a7a67ef5f2bb9b7e930d218e0a8d9976572855bc60bfb0016c4ee061e7eR87):

```js
 return this.recognize(null, utterance, this.predictionOptions);
```

My thought was that if `context` is `undefined` here, there's a bigger issue that I'd want to be caught. That being said, I'm very willing to change it.

Reply:
Leave a comment 😄  then I won't have to remember this explanation.